### DISCUSSION TITLE
How to use DeCollationAnalyzer, but keep the full search capacities

### DISCUSSION THREAD

**Discussion by oiBio** [ravendb/ravendb#15114] [Q&A] [answered]
Hi, 

we want to use the build in CollationAnalyzer with german Collation (DeCollationAnalyzer) so utilize the correct ordering from umlaut in some text fields.

When we add the Analyzer to the index, the ordering works as excepted, but unfortunately we cant use search() anymore.

Example Data

  session.Store(new Model() { Name = "Anton" });
  session.Store(new Model() { Name = "Berta" });
  session.Store(new Model() { Name = "Charlie" });
  session.Store(new Model() { Name = "Oren" });
  session.Store(new Model() { Name = "Özdemir" });         <--------- This is the problematic one
  session.Store(new Model() { Name = "Willhelm" });
  session.Store(new Model() { Name = "Zepelin" });

With the default filter and a simple OrderBy(e => e.Name) "Özdemir" comes last. In the german alphabet it should come after "Oren".
So we added the analyzer and the ordering works. But we cast use search(e => e.Name, "oren") anymore. We just get results, when we match the Name 100% like search(e => e.Name, "Oren")

Is there a way to get both working?

**Comment by ayende** [comment, marked as answer]
Yes, that is correct, you need to index the field *twice*. Once for ordering and once for search.
Basically, the collation is only handling the sorting only, not searching.
